resources/testharness.js

```diff
@@ -937,7 +959,17 @@ policies and contribution forms [3].
                                         String(e),
                                         e && e.stack);
                        tests.complete();
-                   });
+                   })
+            .finally(function()
+                {
+                    tests.promise_setup_pending--;
+                    if (tests.promise_setup_pending === 0) {
+                        var pending = tests.async_test_pending;
+                        tests.async_test_pending = [];
+                        for(var i in pending)
+                            async_test.apply(null, pending[i]);
+                    }
+                });
```

Comment:
I'm going to defer on this mechanism to someone who's more familiar with testharness internals.